aws-api-appsync/src/main/java/com/amplifyframework/api/aws/SelectionSet.java

```diff
@@ -355,18 +355,14 @@ private static boolean isCustomType(@NonNull Field field) {
                 return new HashSet<>();
             }
             Set<SelectionSet> result = new HashSet<>();
-            if (depth == 0 && LeafSerializationBehavior.JUST_ID.equals(requestOptions.leafSerializationBehavior())) {
-                result.add(new SelectionSet("id"));
-                return result;
-            }
             for (Map.Entry<String, ModelField> entry : modelSchema.getFields().entrySet()) {
                 String fieldName = entry.getKey();
                 ModelAssociation association = modelSchema.getAssociations().get(fieldName);
                 if (association != null) {
                     if (depth >= 1) {
                         String associatedModelName = association.getAssociatedType();
-                        ModelSchema associateModelSchema = ModelSchemaRegistry.instance()
-                                .getModelSchemaForModelClass(associatedModelName);
+                        ModelSchema associateModelSchema =
+                            modelSchemaRegistry.getModelSchemaForModelClass(associatedModelName);
```

Comment:
This change necessitated all of the rewiring of the `ModelSchemaRegistry`. `ModelSchemaRegistry.instance()` creates a new empty registry, but we need the instance that got setup by the storage adapter during initialization.